verification/verify_test.go

```diff
@@ -57,19 +61,42 @@ func TestVerify(t *testing.T) {
 	assert.Nil(res)
 	assert.Equal(ErrPixelsAbsent, err)
 
-	// check pixel count fails
+	// check pixel count fails w/ external verifier
 	data.Segments = append(data.Segments, &net.TranscodedSegmentData{Url: "def", Pixels: verifier.results.Pixels[1]})
 	assert.Len(data.Segments, len(verifier.results.Pixels)) // sanity check
 	res, err = sv.Verify(&Params{Results: data})
 	assert.Nil(res)
 	assert.Equal(ErrPixelMismatch, err)
 
-	// Check pixel count succeeds
+	// Check pixel count succeeds w/ external verifier
 	data.Segments[0].Pixels = verifier.results.Pixels[0]
 	res, err = sv.Verify(&Params{Results: data})
 	assert.Nil(err)
 	assert.NotNil(res)
 
+	// check pixel count fails w/o external verifier
+	sv = NewSegmentVerifier(&Policy{Verifier: nil, Retries: 2}) // reset
+	data = &net.TranscodeData{Segments: []*net.TranscodedSegmentData{
+		{Url: "abc", Pixels: verifier.results.Pixels[0] + 1},
+	}}
+	data.Segments = append(data.Segments, &net.TranscodedSegmentData{Url: "def", Pixels: verifier.results.Pixels[1]})
+	assert.Len(data.Segments, len(verifier.results.Pixels)) // sanity check
+	res, err = sv.Verify(&Params{Results: data, Orchestrator: &net.OrchestratorInfo{TicketParams: &net.TicketParams{}}})
+	assert.Nil(res)
+	assert.Equal(ErrPixelMismatch, err)
+
+	// Check pixel count succeeds w/o external verifier
+	pixels, err := pixels("../server/test.flv")
+	assert.Nil(err)
+	data = &net.TranscodeData{Segments: []*net.TranscodedSegmentData{
+		{Url: "../server/test.flv", Pixels: pixels},
+		{Url: "../server/test.flv", Pixels: pixels},
+	}}
+	renditions := [][]byte{{123}, {234}}
+	res, err = sv.Verify(&Params{Results: data, Orchestrator: &net.OrchestratorInfo{TicketParams: &net.TicketParams{}}, Renditions: renditions})
```

Comment:
Technically we don't need `TicketParams` here in this PR, but we'll need it for once #1368 is merged. Maybe omit in this PR, but introduce it in #1368 for the tests needed there (to test sig verification when running `SegmentVerifier.Verify()`).

Reply:
Why not just leave it then, if this PR wont be merged without 1368 ?

Reply:
Mainly for commit history clarity/ease of understanding i.e. introducing a change in the commit where it is actually relevant. Could adjust at the very end, but seems easier to do it now.

Reply:
bd4ecfdd3caf31de7c335f883e50ab659646469b